### PR TITLE
Gangams/aad stage2 full switch to mdsd

### PR DESCRIPTION
**Pull request by ganga1980**
This PR has following changes 
  - conversion of  all fluentd plugins to v1 version
  - upgrade to ruby version 2.6
  - routing all the  CI data types to Linux AMA
  - Linux AMA official release with all the changes 
  - removal of the omsagent dependency
  -  kept hidden fall back option to direct ODS ingestion as v1 for the container log data type for Linux agent
  -  tag names on the out plugins changed to inline aad auth changes related to tag from extension config.
  Note  -> no changes in omsagent.yaml related to fluentd config to keep ability to switch  image tag if required. plan to update this in subsequent releases.
    
Pending items
  - bug bash, fixes for the bugs if there are any during bug bash 
  -  Scale testing and results with current agent vs released candidate

### CODE REVIEW

File: build/common/installer/scripts/tomlparser.rb [hunk 1]
```diff
@@ -162,8 +164,12 @@ def populateSettingValuesFromConfigMap(parsedConfig)
     #Get container logs route setting
     begin
       if !parsedConfig[:log_collection_settings][:route_container_logs].nil? && !parsedConfig[:log_collection_settings][:route_container_logs][:version].nil?
-        @containerLogsRoute = parsedConfig[:log_collection_settings][:route_container_logs][:version]
-        puts "config::Using config map setting for container logs route"
+        if !parsedConfig[:log_collection_settings][:route_container_logs][:version].empty?
```
Comment (vishiy) [inline build/common/installer/scripts/tomlparser.rb:167]: These checks are not needed. The way it used to work is - if the soute is not one of the valid values (including empty), it just uses default.

File: build/common/installer/scripts/tomlparser.rb [hunk 2]
```diff
@@ -256,7 +262,7 @@ def get_command_windows(env_variable_name, env_variable_value)
     file.write(commands)
     commands = get_command_windows('AZMON_CLUSTER_COLLECT_ALL_KUBE_EVENTS', @collectAllKubeEvents)
     file.write(commands)
-    commands = get_command_windows('AZMON_CONTAINER_LOGS_EFFECTIVE_ROUTE', @containerLogsRoute)
+    commands = get_command_windows('AZMON_CONTAINER_LOGS_ROUTE', @containerLogsRoute)
```
Comment (vishiy) [inline build/common/installer/scripts/tomlparser.rb:265]: Is this teste for ADX route ?

File: source/plugins/utils/oms_common.rb [hunk 3]
```diff
@@ -0,0 +1,1020 @@
+module OMS
```
Comment (vishiy) [inline source/plugins/utils/oms_common.rb:1]: why do we need this ? We can just use a file logger and move on.

File: build/linux/installer/conf/container.conf [hunk 4]
```diff
@@ -1,141 +1,168 @@
...
+    retry_max_times 10
+    retry_wait 5s
+    retry_max_interval 5m
+    flush_thread_count 5   
+   </buffer>
+   keepalive true     
+  </match>
+
+  #cadvisorperf
+  <match **LINUX_PERF_BLOB**>
```
Comment (vishiy) [inline build/linux/installer/conf/container.conf:81]: Few things here -
1) if mdsd is not up and running before fluent, fluent will crash. There are ways to customize this.
2) connect & send timeouts need to be specified (not more than 30s for local communication)
3) also better to disable heart beats, as we have seen too many heart beats causing TCP_SYNs
4) what modes have we tested? sync v. async

File: build/linux/installer/datafiles/base_container.data [hunk 5]
```diff
@@ -134,43 +54,128 @@ MAINTAINER:              'Microsoft Corporation'
...
-/opt/microsoft/omsagent/plugin/health/health_monitor_telemetry.rb;                                  source/plugins/ruby/health/health_monitor_telemetry.rb; 644; root; root
-/opt/microsoft/omsagent/plugin/health/health_monitor_utils.rb;                                      source/plugins/ruby/health/health_monitor_utils.rb; 644; root; root
-/opt/microsoft/omsagent/plugin/health/health_signal_reducer.rb;                                     source/plugins/ruby/health/health_signal_reducer.rb; 644; root; root
-/opt/microsoft/omsagent/plugin/health/monitor_factory.rb;                                           source/plugins/ruby/health/monitor_factory.rb; 644; root; root
-/opt/microsoft/omsagent/plugin/health/monitor_set.rb;                                               source/plugins/ruby/health/monitor_set.rb; 644; root; root
-/opt/microsoft/omsagent/plugin/health/unit_monitor.rb;                                              source/plugins/ruby/health/unit_monitor.rb; 644; root; root
+/etc/fluent/plugin/lib/application_insights/version.rb;								   source/plugins/ruby/lib/application_insights/version.rb; 644; root; root
```
Comment (vishiy) [inline build/linux/installer/datafiles/base_container.data:60]: Now that we can gem install can we consume this as gem ?

File: kubernetes/linux/setup.sh [hunk 6]
```diff
@@ -9,37 +9,13 @@ sed -i -e 's/# en_US.UTF-8 UTF-8/en_US.UTF-8 UTF-8/' /etc/locale.gen && \
...
-#/usr/bin/dpkg -i $TMPDIR/omsbundle/100/omsconfig*.deb
-
-#install oneagent - Official bits (05/2021)
-wget https://github.com/microsoft/Docker-Provider/releases/download/05112021-oneagent/azure-mdsd_1.8.0-build.master.189_x86_64.deb
 /usr/bin/dpkg -i $TMPDIR/azure-mdsd*.deb
 cp -f $TMPDIR/mdsd.xml /etc/mdsd.d
 cp -f $TMPDIR/envmdsd /etc/mdsd.d
 
-#Assign permissions to omsagent user to access docker.sock
-sudo apt-get install acl
```
Comment (vishiy) [inline kubernetes/linux/setup.sh:41]: did we check on both docker & containerd nodes before removing ?

File: source/plugins/go/src/oms.go [hunk 7]
```diff
@@ -746,70 +860,144 @@ func PostTelegrafMetricsToLA(telegrafRecords []map[interface{}]interface{}) int
...
+				}
+		}
+		if (len(msgPackEntries) > 0) {							
+				msgpBytes := convertMsgPackEntriesToMsgpBytes(MdsdInsightsMetricsTagName, msgPackEntries)			
+				if MdsdInsightsMetricsMsgpUnixSocketClient == nil {
+					Log("Error::mdsd::mdsd connection does not exist. re-connecting ...")
+					CreateMDSDClientInsightsMetrics(ContainerType)
+					if MdsdInsightsMetricsMsgpUnixSocketClient == nil {
+						Log("Error::mdsd::Unable to create mdsd client for insights metrics. Please check error log.")					
+						ContainerLogTelemetryMutex.Lock()
```
Comment (vishiy) [inline source/plugins/go/src/oms.go:905]: guess you got these from containerlog. Mutexes need to be different for each type.

File: source/plugins/go/src/oms.go [hunk 8]
```diff
@@ -746,70 +860,144 @@ func PostTelegrafMetricsToLA(telegrafRecords []map[interface{}]interface{}) int
...
+				if er != nil {
+					Log("Error::mdsd::Failed to write to mdsd %d records after %s. Will retry ... error : %s", len(msgPackEntries), elapsed, er.Error())
+					if MdsdInsightsMetricsMsgpUnixSocketClient != nil {
+						MdsdInsightsMetricsMsgpUnixSocketClient.Close()
+						MdsdInsightsMetricsMsgpUnixSocketClient = nil
+					}
 
-	jsonBytes, err := json.Marshal(laTelegrafMetrics)
+					ContainerLogTelemetryMutex.Lock()
+					defer ContainerLogTelemetryMutex.Unlock()
```
Comment (vishiy) [inline source/plugins/go/src/oms.go:926]: Can you look into all mutex-es ? I see single mutex used in multiple streams and they are all sourced by different plugins/inputs and each stream will wait on other streams to aquire the lock.

File: source/plugins/go/src/telemetry.go [hunk 9]
```diff
@@ -42,6 +42,10 @@ var (
 	ContainerLogsSendErrorsToMDSDFromFluent float64
 	//Tracks the number of mdsd client create errors for containerlogs (uses ContainerLogTelemetryTicker)
 	ContainerLogsMDSDClientCreateErrors float64
+	//Tracks the number of mdsd client create errors for insightsmetrics (uses ContainerLogTelemetryTicker)
+	InsightsMetricsMDSDClientCreateErrors float64
+	//Tracks the number of mdsd client create errors for kubemonevents (uses ContainerLogTelemetryTicker)
+	KubeMonEventsMDSDClientCreateErrors float64
```
Comment (vishiy) [inline source/plugins/go/src/telemetry.go:48]: where are these telemetry flushed ?

File: source/plugins/go/src/utils.go [hunk 10]
```diff
@@ -119,17 +126,71 @@ func CreateMDSDClient() {
...
+func CreateMDSDClientKubeMon(containerType string) {
+	if MdsdKubeMonMsgpUnixSocketClient != nil {
+		MdsdKubeMonMsgpUnixSocketClient.Close()
+		MdsdKubeMonMsgpUnixSocketClient = nil
+	}
+	/*conn, err := fluent.New(fluent.Config{FluentNetwork:"unix",
+	  FluentSocketPath:"/var/run/mdsd/default_fluent.socket",
+	  WriteTimeout: 5 * time.Second,
+	  RequestAck: true}) */
+	mdsdfluentSocket := "/var/run/mdsd/default_fluent.socket"
```
Comment (vishiy) [inline source/plugins/go/src/utils.go:154]: why is there a client creation code per data type ?

File: source/plugins/go/src/utils.go [hunk 11]
```diff
@@ -119,17 +126,71 @@ func CreateMDSDClient() {
...
+	if err != nil {		
+		Log("Error::mdsd::Unable to open MDSD msgp socket connection for KubeMon events %s",  err.Error())
+		//log.Fatalf("Unable to open MDSD msgp socket connection %s", err.Error())
+	} else {
+		Log("Successfully created MDSD msgp socket connection for KubeMon events:%s", mdsdfluentSocket)
+		MdsdKubeMonMsgpUnixSocketClient = conn
+	}
+}
+
+//mdsdSocketClient to write msgp messages for KubeMonAgent Events
```
Comment (vishiy) [inline source/plugins/go/src/utils.go:169]: we just need one client creation function across all datatypes, as datatype has nothing to do with client creation.

File: source/plugins/ruby/in_kube_nodes.rb [hunk 12]
```diff
@@ -90,6 +97,7 @@ def shutdown
           @condition.signal
         }
         @thread.join
+        super # This super must be at the end of shutdown method
```
Comment (vishiy) [inline source/plugins/ruby/in_kube_nodes.rb:100]: can u explain why super is called in the end ?

File: source/plugins/ruby/in_win_cadvisor_perf.rb [hunk 13]
```diff
@@ -50,11 +53,11 @@ def shutdown
     end
 
     def enumerate()
-      time = Time.now.to_f
+      time = Fluent::Engine.now
```
Comment (vishiy) [inline source/plugins/ruby/in_win_cadvisor_perf.rb:56]: how is this change related to removing omsagent for aad auth? instead of getting time.now you are getting from fluentengine, why ?

File: source/plugins/ruby/in_kube_pvinventory.rb [hunk 14]
```diff
@@ -145,13 +151,8 @@ def parse_and_emit_records(pvInventory, batchTime = Time.utc.iso8601)
...
         records.each do |record|
-          if !record.nil?
-            wrapper = {
-              "DataType" => "KUBE_PV_INVENTORY_BLOB",
-              "IPName" => "ContainerInsights",
-              "DataItems" => [record.each { |k, v| record[k] = v }],
-            }
-            eventStream.add(emitTime, wrapper) if wrapper
+          if !record.nil?          
+            eventStream.add(Fluent::Engine.now, record)
```
Comment (vishiy) [inline source/plugins/ruby/in_kube_pvinventory.rb:155]: please remove all these irrelevant  time changes. Previously we get time once and use it in the emit. Now we are getting time for every record we emit. Even though we dont use emit time this is completely not required as its unnecessary to get time for every single record.,

File: source/plugins/ruby/out_mdm.rb [hunk 15]
```diff
@@ -334,10 +334,10 @@ def send_to_mdm(post_body)
           # Not raising exception, as that will cause retries to happen
         elsif !response.code.empty? && response.code.start_with?("4")
           # Log 400 errors and continue
-          @log.info "Non-retryable HTTPServerException when POSTing Metrics to MDM #{e} Response: #{response}"
+          @log.info "Non-retryable HTTPClientException when POSTing Metrics to MDM #{e} Response: #{response}"
         else
           # raise if the response code is non-400
-          @log.info "HTTPServerException when POSTing Metrics to MDM #{e} Response: #{response}"
+          @log.info "HTTPClientException when POSTing Metrics to MDM #{e} Response: #{response}"
```
Comment (vishiy) [inline source/plugins/ruby/out_mdm.rb:340]: non 400 is server side error.

File: build/linux/installer/conf/container.conf [hunk 16]
```diff
@@ -1,141 +1,168 @@
...
+     retry_max_times 10
+     retry_wait 5s
+     retry_max_interval 5m
+     flush_thread_count 5   
+    </buffer>
+    keepalive true
+  </match>
+
+  <match kubehealth.DaemonSet**>
+    @type health_forward
```
Comment (vishiy) [inline build/linux/installer/conf/container.conf:103]: the main reason health_forward existed is because of on start errors (if the destination [replica] is unreachable duting startup, fluent will crash). Now this has been taken care of in the OOB forward. so you should remove health_forward & use normal forward (and ensure startup errors are configured to be ignored)

File: build/linux/installer/conf/kube.conf [hunk 17]
```diff
@@ -1,270 +1,349 @@
...
-     log_level debug
+     @log_level debug
     </source>
 
     #Kubernetes Persistent Volume inventory
     <source>
-     type kubepvinventory
-     tag oms.containerinsights.KubePVInventory
+     @type kube_pvinventory
+     tag oneagent.containerInsights.KUBE_PV_INVENTORY_BLOB
```
Comment (vishiy) [inline build/linux/installer/conf/kube.conf:20]: So these changes will go into AKS chart for our addon. How is this not a yaml change that will help easy rollback using image ?

File: build/linux/installer/conf/container.conf [hunk 18]
```diff
@@ -1,141 +1,168 @@
...
+    @type containerinventory
+    tag oneagent.containerInsights.CONTAINER_INVENTORY_BLOB
+    run_interval 60
+    @log_level debug
+  </source>
+
+  #cadvisor perf
+  <source>
+    @type cadvisor_perf
+    tag oneagent.containerInsights.LINUX_PERF_BLOB
```
Comment (vishiy) [inline build/linux/installer/conf/container.conf:31]: All these tag changes are unneeded & unrelated to aad/omsagent removal.

**Comment by gracewehner** [comment]
Hi Ganga, how do we update the ruby gems? I ran a trivy scan for cidev:cioneagent05172021-1 and still see these flagged:
image

File: build/linux/installer/scripts/tomlparser-mdm-metrics-config.rb [hunk 19]
```diff
@@ -3,7 +3,7 @@
 
 require_relative "tomlrb"
 require_relative "ConfigParseErrorLogger"
-require_relative "microsoft/omsagent/plugin/constants"
+require_relative "/etc/fluent/plugin/constants"
```
Comment (rashmichandrashekar) [inline build/linux/installer/scripts/tomlparser-mdm-metrics-config.rb:6]: is this intentional? Are we changing the directory path for the plugins?

**Review by ganga1980 — COMMENTED** [review]
> Hi Ganga, how do we update the ruby gems? I ran a trivy scan for cidev:cioneagent05172021-1 and still see these flagged:
> image

> ruby gem installed through gem install as i showed you. since the vulns are in http_parser so I looked at the http_parser gem https://rubygems.org/gems/http_parser.rb/versions to see if we can upgrade to latest version, but version we are using (meaning come with ruby 2.6) is latest version and also there is also active issue https://rubygems.org/gems/http_parser/versions on this.  I think, we will have to wait until new version of the http_parser available.  let me know if we need to reach security team to update on this.

File: build/common/installer/scripts/tomlparser.rb (hunk 1 above)
Comment (ganga1980) [reply build/common/installer/scripts/tomlparser.rb:167]: since the default changed to v2 for Linux &  v1 for windows. In case of Linux, fallback option is v1. This is just safety check to  fall back to defaults if the customer puts as empty as version and also we dont collect the telemetry as empty string. Removing this will impact telemetry not matching with default.

File: build/common/installer/scripts/tomlparser.rb (hunk 2 above)
Comment (ganga1980) [reply build/common/installer/scripts/tomlparser.rb:265]: This just renaming the variable from AZMON_CONTAINER_LOGS_EFFECTIVE_ROUTE to AZMON_CONTAINER_LOGS_ROUTE and shouldnt impact ADX route but I can to test it  out ( or cover as part of bug bash) to ensure nothing impacts. can you please point me the instructions how to validate ADX route?

File: source/plugins/utils/oms_common.rb (hunk 3 above)
Comment (ganga1980) [reply source/plugins/utils/oms_common.rb:1]: We use the  common functions from OMS multiple places like hostname and logger etc.  since this code being already used & validated from the beginning so using the same functions.

File: build/linux/installer/datafiles/base_container.data (hunk 5 above)
Comment (ganga1980) [reply build/linux/installer/datafiles/base_container.data:60]: AI ruby SDK deprecated & archived and there are not accepting prs any more.  there is already an active PR related to the proxy which we would need to use gem approach and in future, if we need any bug fixes and we will not able to use gem approach so we should continue to use the approach what we are using.

File: kubernetes/linux/setup.sh (hunk 6 above)
Comment (ganga1980) [reply kubernetes/linux/setup.sh:41]: Yes, this being validated. we only use docker.sock for the telemetry for the docker version and which is already validated.

File: build/common/installer/scripts/tomlparser.rb (hunk 1 above)
Comment (vishiy) [reply build/common/installer/scripts/tomlparser.rb:167]: whats wrong if telemetry is empty ? your argument just doesn't make sense. telemetry should show whatever the customer puts there.

File: source/plugins/utils/oms_common.rb (hunk 3 above)
Comment (vishiy) [reply source/plugins/utils/oms_common.rb:1]: can we just keep those functions and remove others ?

File: source/plugins/ruby/in_kube_nodes.rb (hunk 12 above)
Comment (ganga1980) [reply source/plugins/ruby/in_kube_nodes.rb:100]: This is upgrade guidance from fluentd. please see examples in this doc - https://docs.fluentd.org/plugin-development/plugin-update-from-v0.12

File: source/plugins/ruby/in_win_cadvisor_perf.rb (hunk 13 above)
Comment (ganga1980) [reply source/plugins/ruby/in_win_cadvisor_perf.rb:56]: > how is this change related to removing omsagent for aad auth? instead of getting time.now you are getting from fluentengine, why ?

> Yes, this is not related to omsagent and aadauth but related to Fluentd upgrade. Here is the guidance https://docs.fluentd.org/plugin-development/plugin-update-from-v0.12 . Time.now doesnt work with fluentdv1 which explictly and also tested.

Comment (vishiy) [reply source/plugins/ruby/in_win_cadvisor_perf.rb:56]: can you do now=Fluent::Engine.now and use that for every record ?

File: source/plugins/ruby/in_kube_pvinventory.rb (hunk 14 above)
Comment (ganga1980) [reply source/plugins/ruby/in_kube_pvinventory.rb:155]: This shouldn't impact anything butI can update to call Fluent::Engine.now once ad use the same pattern as emitTime.

File: build/linux/installer/conf/container.conf (hunk 16 above)
Comment (ganga1980) [reply build/linux/installer/conf/container.conf:103]: as we discussed, since our plan to deprecate health so keeping as is.

File: build/common/installer/scripts/tomlparser.rb (hunk 1 above)
Comment (ganga1980) [reply build/common/installer/scripts/tomlparser.rb:167]: what i am saying is, if customer has the cm with the version="" , we default to v2 but the telemetry will be "" since the telemetry uses the same version what's default.  do you see any issue adding !empty? check?

File: build/linux/installer/conf/kube.conf (hunk 17 above)
Comment (ganga1980) [reply build/linux/installer/conf/kube.conf:20]: > So these changes will go into AKS chart for our addon. How is this not a yaml change that will help easy rollback using image ?
> as I mentioned earlier and also in the description, we will be updating config in kube.conf in omsagent.yaml in subsequenent releases. In main.sh, kube.conf & container.conf will be used by fluentd depending rs vs ds pod. This way, we can roll back to image if required at all.

File: build/linux/installer/conf/container.conf (hunk 18 above)
Comment (ganga1980) [reply build/linux/installer/conf/container.conf:31]: > All these tag changes are unneeded & unrelated to aad/omsagent removal.
> tag change align to extension config from AMA. in case of aad auth mode, tag will be dcr-<guid>:containerinsights:<BLOB_NAME> . the reason for the new tag names to aligin with aad changes to use same configuration.

File: source/plugins/go/src/utils.go (hunk 10 above)
Comment (ganga1980) [reply source/plugins/go/src/utils.go:154]: The reason separate client per data type is to so these clients can be concurrent and indenpendly to talk to socket endpoint.  client creation creates separate client for that data type.

File: source/plugins/ruby/in_win_cadvisor_perf.rb (hunk 13 above)
Comment (ganga1980) [reply source/plugins/ruby/in_win_cadvisor_perf.rb:56]: > can you do now=Fluent::Engine.now and use that for every record ?

Sure, I can make that change.

File: build/linux/installer/conf/kube.conf (hunk 17 above)
Comment (vishiy) [reply build/linux/installer/conf/kube.conf:20]: sorry, dont get it. kube.conf is coming from a configmap which is defined in the yaml. If you make changes to kube.conf, that needs to changed in yaml. So are you saying that the changes you have made now in kube.conf in this PR will not be released for stage2 ?

File: build/linux/installer/conf/container.conf (hunk 18 above)
Comment (vishiy) [reply build/linux/installer/conf/container.conf:31]: DCR doesn;t enforce any name, isn;t it ? you have a mapping between fluent tag and datatype/table in mdsd.xml.

File: build/linux/installer/conf/container.conf (hunk 4 above)
Comment (ganga1980) [reply build/linux/installer/conf/container.conf:81]: thanks for bringing up good points 1) verified fluentd doesnt crash even mdsd not running and mdsd can come up late and fluentd can establish session, but planning to more testing around this to make sure we dont run into issues because of the timing as we learned with incident (2) & (3) will validate with these to make sure these are safe 4) async mode

File: build/linux/installer/conf/container.conf (hunk 18 above)
Comment (ganga1980) [reply build/linux/installer/conf/container.conf:31]: > DCR doesn;t enforce any name, isn;t it ? you have a mapping between fluent tag and datatype/table in mdsd.xml.

> in case of the DCR, tag  discovered from extension config which will be in this format dcr-:containerinsights:<BLOB_NAME> for corresponding data type. guid part is dynamic and associated to configured workspace..

Comment (vishiy) [reply build/linux/installer/conf/container.conf:31]: for our 1p DCR, we decide what we want? are you saying you want to match that ? also where is that , can you add that for reference in a file ?

Comment (ganga1980) [reply build/linux/installer/conf/container.conf:31]: > for our 1p DCR, we decide what we want? are you saying you want to match that ? also where is that , can you add that for reference in a file ?
Here is the example - https://jsoneditoronline.org/#left=local.jufewo&right=local.yepomi. The changes will be next PR in stage 3 roll out.

File: build/linux/installer/conf/kube.conf (hunk 17 above)
Comment (ganga1980) [reply build/linux/installer/conf/kube.conf:20]: > sorry, dont get it. kube.conf is coming from a configmap which is defined in the yaml. If you make changes to kube.conf, that needs to changed in yaml. So are you saying that the changes you have made now in kube.conf in this PR will not be released for stage2 ?
>  what I am saying, in stag2, fluentd will consume the kube.conf which will be available in /etc/fluent/kube.conf path and rather than coming from configmap in the yaml.  once that rolled out, in stage3, either we can keep the kubeconfig configmap in omsagent.yaml keep the sync with kube.conf or we can try the removing of the dependency of the kube.conf configmap dependency from omsagent yaml.

File: source/plugins/utils/oms_common.rb (hunk 3 above)
Comment (ganga1980) [reply source/plugins/utils/oms_common.rb:1]: done.

File: source/plugins/go/src/telemetry.go (hunk 9 above)
Comment (ganga1980) [reply source/plugins/go/src/telemetry.go:48]: thanks for catching and it was miss. fixed

File: source/plugins/go/src/utils.go (hunk 10 above)
Comment (ganga1980) [reply source/plugins/go/src/utils.go]: refactored the code to reuse same create method.

File: source/plugins/go/src/oms.go (hunk 8 above)
Comment (ganga1980) [reply source/plugins/go/src/oms.go:935]: reviewed & looks good to me. we already use the ContainerLogTelemetryMutex for both Telegraf & containerlog related to telemetry. using the same pattern. let me know if I am missing anything which could potential cause deadlock.

File: source/plugins/go/src/oms.go (hunk 7 above)
Comment (ganga1980) [reply source/plugins/go/src/oms.go:914]: reviewed & looks good to me. we already use the ContainerLogTelemetryMutex for both Telegraf & containerlog related to telemetry. using the same pattern. let me know if I am missing anything which could potential cause deadlock.

File: build/common/installer/scripts/tomlparser.rb (hunk 2 above)
Comment (ganga1980) [reply build/common/installer/scripts/tomlparser.rb:265]: verified adx route and looks good to me.

File: build/linux/installer/datafiles/base_container.data [hunk 20]
```diff
@@ -213,11 +208,14 @@ MAINTAINER:              'Microsoft Corporation'
 /opt/td-agent-bit/bin;                      755; root; root;sysdir
 /etc/telegraf;                              755; root; root;sysdir
 
-/opt/microsoft/omsagent/plugin/lib;												755; root; root; sysdir
-/opt/microsoft/omsagent/plugin/lib/application_insights;                        755; root; root; sysdir
-/opt/microsoft/omsagent/plugin/lib/application_insights/channel;                755; root; root; sysdir
-/opt/microsoft/omsagent/plugin/lib/application_insights/channel/contracts;      755; root; root; sysdir
-/opt/microsoft/omsagent/plugin/lib/application_insights/rack;                   755; root; root; sysdir
+/etc/fluent;                                                         755; root; root; sysdir
```
Comment (rashmichandrashekar) [inline build/linux/installer/datafiles/base_container.data:211]: What is the rationale behind changing just the plugin path and keeping the old paths for all the logs?

Comment (ganga1980) [reply build/linux/installer/datafiles/base_container.data:211]: > What is the rationale behind changing just the plugin path and keeping the old paths for all the logs?

> plugin path changed because of the omsagent removal and log path also changed. When you say "old paths for all the logs", can you please point me which log paths you are referring here?

Comment (ganga1980) [reply build/linux/installer/datafiles/base_container.data:211]: only change with regards to log paths for the ones which are using /var/opt/microsoft/**omsagent**/log path. This change because of the omsagent removal. These will be moved under /var/opt/microsoft/dockercim-prov/log path.

File: kubernetes/linux/setup.sh [hunk 21]
```diff
@@ -9,37 +9,13 @@ sed -i -e 's/# en_US.UTF-8 UTF-8/en_US.UTF-8 UTF-8/' /etc/locale.gen && \
     dpkg-reconfigure --frontend=noninteractive locales && \
     update-locale LANG=en_US.UTF-8
 
-wget https://github.com/Microsoft/OMS-Agent-for-Linux/releases/download/OMSAgent_v1.10.0-1/omsagent-1.10.0-1.universal.x64.sh
+#install oneagent - Official bits (05/17/2021)
+wget https://github.com/microsoft/Docker-Provider/releases/download/05172021-oneagent/azure-mdsd_1.10.1-build.master.213_x86_64.deb
```
Comment (daweim0) [inline kubernetes/linux/setup.sh:13]: Have we scale tested mdsd 1.10? (I think I tested 1.8)

**Review by daweim0 — APPROVED** [review]

File: kubernetes/linux/setup.sh (hunk 21 above)
Comment (ganga1980) [reply kubernetes/linux/setup.sh:13]: Will be testing after merging to cidev
